hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/Route.java

```diff
@@ -236,8 +238,32 @@ private MQQueueConnectionFactory mqQueueConnectionFactory()  {
         	mqQueueConnectionFactory.setPort(Integer.valueOf(properties.getValue(MQ_PORT)));
         	mqQueueConnectionFactory.setQueueManager(properties.getValue(MQ_QUEUEMANAGER));
     		logger.debug("{} - MQ connection factory has been created.", methodName);			
-        } catch (JMSException jmse) {
-        	 logger.error("{} - MQ connection factory initialization failed with the error : {}", methodName, jmse.getMessage());       	
+        } catch (JMSException je) {
+        	// Check for linked exceptions in JMSException
+            Throwable t = je;
+            while (t != null) {          
+                // Add on specific information depending on the type of exception
+                if (t instanceof JMSException) {
+                    JMSException je1 = (JMSException) t;
+                    logger.error("{} - JMS Error code: {}", methodName, je1.getErrorCode());
+                } else if (t instanceof MQException) {
+                    MQException mqe = (MQException) t;
+                    logger.error("{} - WMQ Completion code: {}" , methodName, mqe.getCompCode());
+                    logger.error("{} - WMQ Reason code: {}" , methodName, mqe.getReason());
+                } else if (t instanceof JmqiException){
+                    JmqiException jmqie = (JmqiException)t;
+                    logger.error("{} - WMQ Log Message: {}", methodName, jmqie.getWmqLogMessage());
+                    logger.error("{} - WMQ Explanation: {}", methodName, jmqie.getWmqMsgExplanation());
+                    logger.error("{} - WMQ Msg Summary: {}", methodName, jmqie.getWmqMsgSummary());
+                    logger.error("{} - WMQ Msg User Response: "
+                                       + jmqie.getWmqMsgUserResponse());
+                    logger.error("{} - WMQ Msg Severity: {}", methodName, jmqie.getWmqMsgSeverity());
+                }
+                
+                // Get the next cause
+                t = t.getCause();
+                logger.error("{} - MQ connection factory initialization failed with the error : {}", methodName, t.getMessage());
```

Comment:
I believe this line will cause a NPE once there are no more causes. I think this general log statement should be outside the logic for handling specific exceptions, i.e. the first line in the catch clause.